src/main/java/dev/jbang/cli/BaseBuildCommand.java

```diff
@@ -90,6 +90,7 @@ static Source buildIfNeeded(Source src, RunContext ctx) throws IOException {
 	static Source build(ScriptSource src, RunContext ctx) throws IOException {
 		Source result = src;
 
+		// todo: pass properties through instead of changing system properties
```

Comment:
i noticed this and similar in alias where there is a reliance on system properties having been set upfront. That doesn't work well if want to have this as a library as it would then be part of global state.

Thus it is not relevant to this specific issue but just took note of it.

Reply:
i went ahead and fixed property passing so now runcontext have a getContextProperties() which is backed by system properties rather than require changing system properties.